src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -143,8 +143,8 @@ def build_public_ip_resource(cmd, name, location, tags, address_allocation, dns_
         'dependsOn': [],
         'properties': public_ip_properties
     }
-    if zone:
-        public_ip['zones'] = zone
+    if zone and len(zone) == 1:
+        public_ip['zones'] = zone  # PIP itself doesn't support multiple zones. Standard LB will handle it
```

Comment:
Why is this needed?  If zone doesn't have nargs='+' then how would it ever be >1?  What happens if it is >1?  Right now it looks like it would ignore it. If you somehow get that condition and it is invalid, then an error should be thrown.

Reply:
Please update the comment to reflect our discussion.

Reply:
Done